File: backend/src/bond_monitor/domain/trading/suggestions.py
"""Trading suggestion read-model."""

from __future__ import annotations

from dataclasses import dataclass
from datetime import date
from typing import Literal

SuggestionKind = Literal["buy", "reinvest", "reinvest_watch", "put_offer_reminder", "put_offer_watch", "sell"]


@dataclass(frozen=True)
class Suggestion:
    """Рекомендация к действию (не persisted pending)."""

    id: str
    kind: SuggestionKind
    isin: str
    name: str
    lots: int
    figi: str | None
    suggested_price_pct: float | None
    reason: str
    market_price_pct: float | None = None
    due_date: date | None = None
    source_isin: str | None = None
    chat_template: str | None = None
    urgency: Literal["normal", "soon", "critical"] = "normal"
    risk_acknowledgeable: bool = False
    offer_window_status: str | None = None
    submission_start: date | None = None
    submission_end: date | None = None

File: backend/internal/domain/trading/suggestions.go
package trading

import "time"

// SuggestionKind classifies trading recommendations for the UI queue.
type SuggestionKind string

const (
	SuggestionKindBuy              SuggestionKind = "buy"
	SuggestionKindReinvest         SuggestionKind = "reinvest"
	SuggestionKindReinvestWatch      SuggestionKind = "reinvest_watch"
	SuggestionKindPutOfferReminder   SuggestionKind = "put_offer_reminder"
	SuggestionKindPutOfferWatch      SuggestionKind = "put_offer_watch"
	SuggestionKindSell               SuggestionKind = "sell"
)

// SuggestionUrgency drives UI ordering and Telegram policy.
type SuggestionUrgency string

const (
	SuggestionUrgencyNormal   SuggestionUrgency = "normal"
	SuggestionUrgencySoon     SuggestionUrgency = "soon"
	SuggestionUrgencyCritical SuggestionUrgency = "critical"
)

// Suggestion is an actionable or informational trading recommendation.
type Suggestion struct {
	ID                   string
	Kind                 SuggestionKind
	ISIN                 string
	Name                 string
	Lots                 int
	FIGI                 *string
	SuggestedPricePct    *float64
	Reason               string
	MarketPricePct       *float64
	DueDate              *time.Time
	SourceISIN           *string
	ChatTemplate         *string
	Urgency              SuggestionUrgency
	RiskAcknowledgeable  bool
	OfferWindowStatus    *string
	SubmissionStart      *time.Time
	SubmissionEnd        *time.Time
}
